test/e2e/redundancy/gateway_failover.go

```diff
@@ -137,118 +135,71 @@ func defaultEndpointType() tcp.EndpointType {
 }
 
 func testGatewayFailOverScenario(f *subFramework.Framework) {
-	clusterAName := framework.TestContext.ClusterIDs[framework.ClusterA]
-	clusterBName := framework.TestContext.ClusterIDs[framework.ClusterB]
+	primaryCluster := f.FindClusterWithMultipleGateways()
+	secondaryCluster := framework.FindOtherClusterIndex(primaryCluster)
 
-	nonGatewayNodes := f.FindNodesByGatewayLabel(framework.ClusterA, false)
-	if len(nonGatewayNodes) == 0 {
-		framework.Skipf("Skipping the test as cluster %q doesn't have any suitable non-gateway nodes...", clusterAName)
+	if primaryCluster == -1 || secondaryCluster == -1 {
+		framework.Skipf("Skipping the test as two gateways nodes not found on any cluster...")
```

Comment:
We require at least 2 clusters so no need to check `secondaryCluster`.

```suggestion
	if primaryCluster == -1 {
	    framework.Skipf("No cluster found with multiple gateways, skipping the fail-over test...")
	    return
	}
```

Reply:
Done